pkg/resources/istiod/deployment.go

```diff
@@ -205,6 +215,63 @@ func (r *Reconciler) containers() []apiv1.Container {
 	return containers
 }
 
+func (r *Reconciler) vaultENVInitContainer() apiv1.Container {
+	vaultEnvContainer := apiv1.Container{
+		Name:  "vault-env",
+		Image: util.PointerToString(r.Config.Spec.Istiod.CA.Vault.VaultEnvImage),
+		Args: []string{
+			"vault-env",
+			"sh",
+			"-c",
+			`echo "$ISTIO_CA_CERT" > ca-cert.pem; echo "$ISTIO_CA_CERT_CHAIN" > root-cert.pem; echo -e "$ISTIO_CA_CERT\n$ISTIO_CA_CERT_CHAIN" > cert-chain.pem; echo "$ISTIO_CA_KEY" > ca-key.pem`,
+		},
+		Env: []apiv1.EnvVar{
+			{
+				Name:  "VAULT_ADDR",
+				Value: util.PointerToString(r.Config.Spec.Istiod.CA.Vault.Address),
+			},
+			{
+				Name:  "VAULT_ROLE",
+				Value: util.PointerToString(r.Config.Spec.Istiod.CA.Vault.Role),
+			},
+			{
+				Name:  "VAULT_CACERT",
+				Value: "/vault/tls/ca.crt",
+			},
+			{
+				Name:  "ISTIO_CA_CERT",
+				Value: util.PointerToString(r.Config.Spec.Istiod.CA.Vault.CertPath),
+			},
+			{
+				Name:  "ISTIO_CA_KEY",
+				Value: util.PointerToString(r.Config.Spec.Istiod.CA.Vault.KeyPath),
+			},
+		},
+		WorkingDir: "/etc/cacerts",
+		VolumeMounts: []apiv1.VolumeMount{
+			{
+				MountPath: "/etc/cacerts",
+				Name:      "cacerts",
+			},
+			{
+				MountPath: "/vault/tls",
+				Name:      "vault-tls",
+			},
+		},
+	}
+
+	certChainPath := util.PointerToString(r.Config.Spec.Istiod.CA.Vault.CertPath)
+	if r.Config.Spec.Istiod.CA.Vault.CertChainPath != nil {
+		certChainPath = util.PointerToString(r.Config.Spec.Istiod.CA.Vault.CertChainPath)
+	}
+	vaultEnvContainer.Env = append(vaultEnvContainer.Env, apiv1.EnvVar{
+		Name:  "ISTIO_CA_CERT_CHAIN",
+		Value: certChainPath,
+	})
```

Comment:
Nit: `certChainPath` could be assigned to its appropriate value before initializing the `vaultEnvContainer` variable, that way the `ISTIO_CA_CERT_CHAIN` env var could be added to `vaultEnvContainer.Env` during initialization instead of appending to it later.

Reply:
👍